### PR TITLE
Update Mithril architecture documentation

### LINKED ISSUE #1488: Update Mithril network architecture documentation
## Why
Some users in the community raised concerns about architecture documentation being misleading and hard to understand. 

## What
Update and simplify the architecture diagrams on the Mithril network architecture page on the documentation website: https://mithril.network/doc/mithril/mithril-network/architecture

## How

- [x] Update the architecture diagrams
- [x] Update the documentation page for architecture

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes:
- A C4 diagram for the Mithril network architecture
- And the update of the architecture page of the documentation 

Screenshot from 2024-02-12 18-09-09

## Pre-submit checklist

- Branch
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update documentation website (if relevant)

## Issue(s)
Closes #1488

### CODE REVIEW

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+@startuml architecture
+title "Mithril network architecture"
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+AddElementTag("external", $bgColor="#c1c1c1")
+
+Person(end_user, "End user", "Retrieves and verifies artifacts certified by the Mithril network")
+
+Container_Boundary(mithril_aggregator, "Mithril aggregator", $link="https://github.com/input-output-hk/mithril/tree/main/mithril-aggregator") {
+    Component(mithril_aggregator_runtime, "Aggregator Runtime", "FSM", "Maintains and transitions internal states of aggregator")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:10]: Is 'Runtime' a proper name? If not, suggest using 'Aggregator runtime'.
In this sentence 'Maintains and transitions internal states of aggregator' suggest either:
'Maintains and transitions aggregator's internal states' or 'Maintains and transitions internal states of the aggregator'

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_aggregator_cryptographic_engine, "Cryptographic Engine", "STM/Asym", "Operates cryptographic primitives")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:11]:
```suggestion
    Component(mithril_aggregator_cryptographic_engine, "Cryptographic engine", "STM/Asym", "Operates cryptographic primitives")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_aggregator_http_server, "HTTP server", "REST API", "Serves HTTP requests to signers and clients")
+    ComponentDb(mithril_aggregator_persistence, "Persistence", "SQLite/FS", "Stores aggregator data (verification keys, certificates, artifacts)")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:13]:
```suggestion
    ComponentDb(mithril_aggregator_persistence, "Persistence", "SQLite/FS", "Stores aggregator's data (verification keys, certificates, artifacts)")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    
+    Rel(mithril_aggregator_runtime, mithril_aggregator_cryptographic_engine, "Computes protocol cryptography")
+    BiRel(mithril_aggregator_runtime, mithril_aggregator_persistence, "Access aggregator state")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:16]:
```suggestion
    BiRel(mithril_aggregator_runtime, mithril_aggregator_persistence, "Accesses aggregator's state")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    BiRel(mithril_aggregator_http_server, mithril_aggregator_persistence, "Access aggregator state")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:17]:
```suggestion
    BiRel(mithril_aggregator_http_server, mithril_aggregator_persistence, "Accesses aggregator's state")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+Container_Boundary(mithril_signer, "Mithril signer", $link="https://github.com/input-output-hk/mithril/tree/main/mithril-signer") {
+    Component(mithril_signer_runtime, "Signer Runtime", "FSM", "Maintains and transitions internal states of signer")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:21]:
```suggestion
    Component(mithril_signer_runtime, "Signer Runtime", "FSM", "Maintains and transitions signer's internal states ")
```

Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:21]: Same question about 'Runtime', if possible, change to 'runtime'

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_signer_cryptographic_engine, "Cryptographic Engine", "STM/Asym", "Operates cryptographic primitives")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:22]:
```suggestion
    Component(mithril_signer_cryptographic_engine, "Cryptographic engine", "STM/Asym", "Operates cryptographic primitives")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_signer_http_client, "HTTP client", "REST API", "Sends HTTP requests to aggregator via relay")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:23]:
```suggestion
    Component(mithril_signer_http_client, "HTTP client", "REST API", "Sends HTTP requests to the aggregator via the relay")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    ComponentDb(mithril_signer_persistence, "Persistence", "SQLite/FS", "Stores signer data (signing keys)")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:24]:
```suggestion
    ComponentDb(mithril_signer_persistence, "Persistence", "SQLite/FS", "Stores signer's data (signing keys)")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+    BiRel(mithril_signer_runtime, mithril_signer_http_client, "Retrieves aggregator data")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:26]:
```suggestion
    BiRel(mithril_signer_runtime, mithril_signer_http_client, "Retrieves aggregator's data")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    Rel(mithril_signer_runtime, mithril_signer_cryptographic_engine, "Computes protocol cryptography")
+    BiRel(mithril_signer_runtime, mithril_signer_persistence, "Access signer state")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:28]:
```suggestion
    BiRel(mithril_signer_runtime, mithril_signer_persistence, "Accesses signer's state")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+Container_Boundary(mithril_relay, "Mithril relay") {
+    Component(mithril_relay_forward_proxy, "Forward proxy", "Squid", "Proxies HTTPS calls to aggregator")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:32]:
```suggestion
    Component(mithril_relay_forward_proxy, "Forward proxy", "Squid", "Proxies HTTPS calls to the aggregator")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+Container_Boundary(mithril_client, "Mithril client", $link="https://github.com/input-output-hk/mithril/tree/main/mithril-client") {
+    Component(mithril_client_runtime, "Client Runtime", "Client commands", "Retrieves artifacts and verifies their authenticity")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:36]: 'runtime'?

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_client_cryptographic_engine, "Cryptographic Engine", "STM/Asym", "Operates cryptographic primitives")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:37]:
```suggestion
    Component(mithril_client_cryptographic_engine, "Cryptographic engine", "STM/Asym", "Operates cryptographic primitives")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    Component(mithril_client_http_client, "HTTP client", "REST API", "Sends HTTP requests to aggregator")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:38]:
```suggestion
    Component(mithril_client_http_client, "HTTP client", "REST API", "Sends HTTP requests to the aggregator")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+
+    BiRel(end_user, mithril_client_runtime, "CLI or browser", "User interaction")
+    BiRel(mithril_client_runtime, mithril_client_http_client, "Retrieves aggregator data and artifacts")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:41]:
```suggestion
    BiRel(mithril_client_runtime, mithril_client_http_client, "Retrieves aggregator's data and artifacts")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    Rel(mithril_client_runtime, mithril_client_cryptographic_engine, "Computes protocol cryptography")
+}
+
+Rel(mithril_signer_http_client, mithril_relay_forward_proxy, "", "JSON/TCP")
+Rel(mithril_relay_forward_proxy, mithril_aggregator_http_server, "", "JSON/HTTPS")
+Rel(mithril_client_http_client, mithril_aggregator_http_server, "", "JSON/HTTPS")
+
+Component(mithril_aggregator_cardano_node, "Cardano Node", "N2C/FS", "Cardano full node", $tags="external")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:49]:
```suggestion
Component(mithril_aggregator_cardano_node, "Cardano node", "N2C/FS", "Cardano full node", $tags="external")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+Rel_L(mithril_aggregator_runtime, mithril_aggregator_cardano_node, "Retrieve Cardano chain state")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:50]:
```suggestion
Rel_L(mithril_aggregator_runtime, mithril_aggregator_cardano_node, "Retrieves Cardano chain state")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+Component(mithril_signer_cardano_node, "Cardano Node", "N2C/FS", "Cardano block producer node", $tags="external")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:51]:
```suggestion
Component(mithril_signer_cardano_node, "Cardano node", "N2C/FS", "Cardano block-producing node", $tags="external")
```

File: docs/website/root/mithril/mithril-network/architecture-c4.puml [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+Rel(mithril_signer_runtime, mithril_signer_cardano_node, "Retrieve Cardano chain state")
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture-c4.puml:52]:
```suggestion
Rel(mithril_signer_runtime, mithril_signer_cardano_node, "Retrieves Cardano chain state")
```

File: docs/website/root/mithril/mithril-network/architecture.md [hunk 21]
```diff
@@ -5,43 +5,38 @@ sidebar_label: Architecture
 
 # Mithril network architecture
 
-:::tip
+:::info
 
-For more information about the Mithril protocol, read [about Mithril](../mithril-protocol/protocol.md).
+* The current Mithril network is based on a single aggregator. However, the team is currently working on a more decentralized version where multiple aggregators will be operating on the same Mithril network.
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture.md:10]:
```suggestion
* The current Mithril network relies on a single aggregator. However, the team is working on a more decentralized version, where multiple aggregators will operate within the same Mithril network.
```

File: docs/website/root/mithril/mithril-network/architecture.md [hunk 22]
```diff
@@ -5,43 +5,38 @@ sidebar_label: Architecture
...
 * [**Mithril client**](./client.md):
 
-> The node used to restore a Cardano full node by retrieving a remote snapshot and its certificate chain from a Mithril aggregator. It then employs Mithril cryptographic primitives to verify their authenticity and validity.
+> The node used to restore artifacts produced by a Mithril aggregator. It then employs Mithril cryptographic primitives to verify their authenticity and validity.
 
-## Architecture overview
+* **Mithril relay**:
 
-:::info
+> A forward proxy which is used to secure communications between the Mithril signer and the Mithril aggregator. More information are available at the [Mithril signer deployment model](../../manual/getting-started/run-signer-node#mithril-signer-deployment-model) section
```
Comment (olgahryniuk) [inline docs/website/root/mithril/mithril-network/architecture.md:36]:
```suggestion
> A forward proxy that is used to secure communications between the Mithril signer and the Mithril aggregator. More information is available in the [Mithril signer deployment model](../../manual/getting-started/run-signer-node#mithril-signer-deployment-model) section.
```

**Review by Alenar — APPROVED** [review]

**Review by dlachaumepalo — APPROVED** [review]

**Pull request merged**